### PR TITLE
Some improvements for Zypper `dist-upgrade`

### PR DESCRIPTION
**Pull request by meaksh**
### What does this PR do?

This PR adds a few extra improvements for the zypper `pkg.upgrade`.
1. `fromrepo` now accepts multiple repositories
2. `dryrun` returns the zypper output
3. `novendorchange` is not supported in SLE11 zypper version.
### Tests written?

Yes

/cc @isbm

### CODE REVIEW

File: salt/modules/zypper.py [hunk 1]
```diff
@@ -1121,23 +1121,30 @@ def upgrade(refresh=True,
...
-            log.info('Targeting repo {0!r}'.format(fromrepo))
+            for repo in fromrepo:
+                cmd_update.extend(['--from', repo])
+            log.info('Targeting repos: {0!r}'.format(fromrepo))
 
         if novendorchange:
-            cmd_update.append('--no-allow-vendor-change')
-            log.info('Disabling vendor changes')
+            # TODO: Grains validation should be moved to Zypper class
+            if __grains__['osrelease_info'][0] != 11:
```
Comment (isbm) [inline salt/modules/zypper.py]: '>', not '!=' :-)

**Review by isbm — CHANGES_REQUESTED** [review]
Please make sure it is applicable only to SLE12 :-)

File: salt/modules/zypper.py (hunk 1 above)
Comment (meaksh) [inline salt/modules/zypper.py]: You're right! Better to prevent. Thanks! :)

**Review by isbm — APPROVED** [review]
Great, thanks!

**Pull request merged**